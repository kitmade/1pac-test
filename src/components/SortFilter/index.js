import React from "react";
import PropTypes from "prop-types";

const SortFilter = (props) => {
  const { filters, onSelected } = props;

  const onOptionSelect = (e) => {
    const { value } = e.target;
    const { orderBy, sort } = filters.filter(
      (filter) => filter.title === value
    )[0];
    const cond = { orderBy, sort };
    if (typeof onSelected === "function") {
      onSelected(cond);
    }
  };

  return (
    <select onChange={onOptionSelect}>
      {Array.isArray(filters) &&
        filters.map((item, index) => (
          <option key={`filer-${index}`} value={item.title}>
            {item.title}
          </option>
        ))}
    </select>
  );
};

SortFilter.propTypes = {
  onSelected: PropTypes.func,
  filter: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      orderBy: PropTypes.string,
      sort: PropTypes.string,
    })
  ),
};

export default SortFilter;
