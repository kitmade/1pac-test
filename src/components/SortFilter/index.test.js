import React from "react";
import { render, fireEvent, getAllByText } from "@testing-library/react";
import renderer from "react-test-renderer";
import SortFilter from ".";

const defaultData = [
  {
    title: "test-asc",
    orderBy: "test",
    sort: "asc",
  },
  {
    title: "test-desc",
    orderBy: "test",
    sort: "desc",
  },
];

describe("SortFilter", () => {
  let filters = [];

  beforeEach(() => {
    filters = defaultData;
  });

  it("should match snapshot", () => {
    const comp = renderer.create(<SortFilter filters={filters} />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it("should match snapshot with filters is empty", () => {
    const comp = renderer.create(<SortFilter filters={[]} />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it("should be able to select an option and invoke onSelected", () => {
    const mockOnSelected = jest.fn();
    const { container, getByText } = render(
      <SortFilter filters={filters} onSelected={mockOnSelected} />
    );
    fireEvent.click(container.querySelector("select"));
    fireEvent.click(getByText("test-desc"));
    fireEvent.change(container.querySelector("select"), {
      target: { value: "test-desc" },
    });
    expect(mockOnSelected).toHaveBeenCalled();
    expect(mockOnSelected).toBeCalledWith({
      orderBy: "test",
      sort: "desc",
    });
  });
});
