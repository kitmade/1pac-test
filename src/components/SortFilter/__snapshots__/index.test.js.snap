// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortFilter should match snapshot 1`] = `
<select
  onChange={[Function]}
>
  <option
    value="test-asc"
  >
    test-asc
  </option>
  <option
    value="test-desc"
  >
    test-desc
  </option>
</select>
`;

exports[`SortFilter should match snapshot with filters is empty 1`] = `
<select
  onChange={[Function]}
/>
`;
